woocommerce/wc-admin
# Refactor profile wizard benefits step and add tests

Fixes #6020 

* Refactors the benefits step with a more declarative `isInstalling` state.
* Allows visiting the benefits page when plugins have been installed but not connected.
* Redirects to the homescreen if Jetpack is installed and all plugins are connected.
* Adds tests.

Note that the "Yes please!" does not really work to connect to Jetpack on the 2nd go round if both plugins are already activated.  Should we update this to allow this if they directly visit this page?  (I'm think yes, but need a sanity check). cc @pmcpinto @elizaan36 

### Screenshots

#### Before
<img width="393" alt="Screen Shot 2021-03-12 at 10 38 25 AM" src="https://user-images.githubusercontent.com/10561050/111009905-dcd65600-8362-11eb-8794-68c28ddcec19.png">


#### After
<img width="835" alt="Screen Shot 2021-03-12 at 6 43 30 PM" src="https://user-images.githubusercontent.com/10561050/111009906-dfd14680-8362-11eb-99a5-de4e1ecc8524.png">


### Detailed test instructions:

1. Deactivate Jetpack and/or WooCommerce Services.
2. Visit the profiler benefits page. `/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=benefits`
3. Click "Yes please!" to continue.
4. Without connecting to Jetpack, navigate backwards using your browser's back button.
5. Make sure the page continues to display (benefits may have changed) and that action buttons are functional.
6. Make sure skipping the install works as expected.
7. Connect Jetpack.
8. Attempt to directly visit the benefits page. `/wp-admin/admin.php?page=wc-admin&path=%2Fsetup-wizard&step=benefits`
9. Note that you are redirected to the homescreen.
